registration/migrations/0007_fix_invalid_domain_names.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2018-01-05 17:04
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import Func, F, Value
+from django.db.models.functions import Lower
+
+
+def clean_domain_name(field_name):
+    """
+    Returns a Django query expression that replaces all underscores with a
+    hyphen, removes leading and trailing hyphens, and converts the field to
+    lower case.
+    """
+    remove_underscores = Func(
+        F(field_name),
+        Value('_'),
+        Value('-'),
+        function='replace'
+    )
+    trim_hyphens = Func(
+        remove_underscores,
+        Value('-'),
+        function='btrim',
+    )
+    remove_trailing_hyphens = Func(
+        trim_hyphens,
+        Value(r'[-]+\.'),
+        Value('.'),
+        Value('g'),
+        function='regexp_replace'
+    )
+    return Lower(remove_trailing_hyphens)
```

Comment:
Good that you're not stripping out `.`s here -- that would break `internal_lms_domain` 👍

Reply:
Yes, that is why I left it out. A  dot in internal_lms_domain is valid anyway. I think the registration subdomain case will be taken care of when we change the subdomain manually in the CMS based on feedback (if any).

Reply:
Yep 👍